### PR TITLE
Showable logging functions

### PR DESCRIPTION
**Pull request by bitemyapp**
Used SH for SHowable. Hopefully this is good enough.

### CODE REVIEW

File: monad-logger/Control/Monad/Logger.hs [hunk 1]
```diff
@@ -270,6 +282,29 @@ logError = logTH LevelError
 logOther :: Text -> Q Exp
 logOther = logTH . LevelOther
 
+
+-- | Generates a function that takes a 'Show a => a' and logs a 'LevelDebug' message. Usage:
+--
+-- > $(logDebug) "This is a debug log message"
```
Comment (snoyberg) [inline monad-logger/Control/Monad/Logger.hs]: Can you add `@since` comments and do a minor version bump in the cabal file/changelog?

Comment (bitemyapp) [inline monad-logger/Control/Monad/Logger.hs]: Let me know if this is satisfactory please :)

File: monad-logger/Control/Monad/Logger.hs [hunk 2]
```diff
@@ -270,6 +285,31 @@ logError = logTH LevelError
...
+logWarnSH = logTHShow LevelWarn
+-- | See 'logDebugSH'
+logErrorSH :: Q Exp
+logErrorSH = logTHShow LevelError
+
+-- | Generates a function that takes a 'Show a => a' and logs a 'LevelOther' message. Usage:
+--
+-- > $(logOtherSH "My new level") "This is a log message"
+logOtherSH :: Text -> Q Exp
+logOtherSH = logTH . LevelOther
```
Comment (snoyberg) [inline monad-logger/Control/Monad/Logger.hs]: Is this supposed to be `logTHShow`?

Comment (bitemyapp) [inline monad-logger/Control/Monad/Logger.hs]: Sigh, yes, good catch. Fixed. Sorry about that :\

Comment (snoyberg) [inline monad-logger/Control/Monad/Logger.hs]: NP, that's what code review is for :)

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**